tiledb/sm/fragment/offsets_fragment_metadata.h

```diff
@@ -0,0 +1,736 @@
+/**
+ * @file  offsets_fragment_metadata.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2024 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines class OffsetsFragmentMetadata.
+ */
+
+#ifndef TILEDB_OFFSETS_FRAGMENT_METADATA_H
+#define TILEDB_OFFSETS_FRAGMENT_METADATA_H
+
+#include <deque>
+#include <mutex>
+#include <unordered_map>
+#include <vector>
+
+#include "tiledb/common/common.h"
+#include "tiledb/common/pmr.h"
+#include "tiledb/sm/array_schema/array_schema.h"
+#include "tiledb/sm/filesystem/uri.h"
+#include "tiledb/sm/misc/types.h"
+#include "tiledb/sm/rtree/rtree.h"
```

Comment:
Let's find another name for this class as it includes the RTree which is not offsets data.